source/vir/src/sst_to_air.rs

```diff
@@ -742,10 +804,57 @@ fn exp_to_bv_expr(state: &State, exp: &Exp) -> Expr {
                     "IntRange error: should be I(_) or U(_) for bit-vector, got {:?}",
                     exp.typ
                 ),
-                UnaryOp::Trigger(_) => panic!("Trigger is not supported in bit-vector"),
+                UnaryOp::Trigger(_) => exp_to_bv_expr(exp),
             }
         }
-        _ => panic!("unhandled bv expr conversion {:?}", exp.x),
+        ExpX::UnaryOpr(op, exp) => match op {
+            UnaryOpr::Box(_) | UnaryOpr::Unbox(_) => {
+                return exp_to_bv_expr(exp);
+            }
+            _ => panic!("unsupported for bit-vector: UnaryOpr {:?}", op),
+        },
+        ExpX::If(cond, tbranch, fbranch) => {
+            let cond = exp_to_bv_expr(&cond);
+            let bv_tbranch = exp_to_bv_expr(&tbranch);
+            let bv_fbranch = exp_to_bv_expr(&fbranch);
+            return Arc::new(ExprX::IfElse(cond, bv_tbranch, bv_fbranch));
+        }
```

Comment:
The `ExpX::If` and `ExpX::Bind` look like duplicates of what's in `exp_to_expr`.  Can we merge `exp_to_bv_expr` and `exp_to_expr` together so that we don't duplicate these cases?  Here's what I'd suggest.  `exp_to_expr` already has a special `ExprCtxt` that gets passed around.  You can redefine `ExprCtxt` so that it is a struct with a boolean value saying whether we're processing a bit vector expression or not.  Then the match arms can test this boolean value, as in `ExpX::Unary(op, exp) if expr_ctxt.is_bit_vector_mode => { ... }`.

Reply:
I see, I agree that this change would make codes look much better. Thanks!